ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+cluster_profile=/var/run/secrets/ci.openshift.io/cluster-profile
+cluster_name=${NAMESPACE}-${JOB_NAME_HASH}
+
+out=/tmp/secret/install-config.yaml
+mkdir "$(dirname "${out}")"
+
+cluster_variant=
+if [[ -e "${SHARED_DIR}/install-config-variant.txt" ]]; then
+    cluster_variant=$(<"${SHARED_DIR}/install-config-variant.txt")
+fi
+
+base_domain=
+if [[ -e "${SHARED_DIR}/install-config-base-domain.txt" ]]; then
+    base_domain=$(<"${SHARED_DIR}/install-config-base-domain.txt")
+else
+    case "${CLUSTER_TYPE}" in
+    aws) base_domain=origin-ci-int-aws.dev.rhcloud.com;;
+    azure) base_domain=ci.azure.devcluster.openshift.com;;
+    gcp) base_domain=origin-ci-int-gce.dev.openshift.com;;
+    esac
+fi
+
+echo "Installing from release ${RELEASE_IMAGE_LATEST}"
+
+expiration_date=$(date -d '4 hours' --iso=minutes --utc)
+ssh_pub_key=$(<"${cluster_profile}/ssh-publickey")
+pull_secret=$(<"${cluster_profile}/pull-secret")
+
+workers=3
+if [[ "${cluster_variant}" =~ "compact" ]]; then
+    workers=0
+fi
+
+case "${CLUSTER_TYPE}" in
+aws)
+    aws_region=us-east-1
+    master_type=null
+    if [[ "${cluster_variant}" =~ "xlarge" ]]; then
+        master_type=m5.8xlarge
+    elif [[ "${cluster_variant}" =~ "large" ]]; then
+        master_type=m5.4xlarge
+    fi
+    subnets="[]"
+    if [[ "${cluster_variant}" =~ "shared-vpc" ]]; then
+        case $((RANDOM % 4)) in
+        0) subnets="['subnet-030a88e6e97101ab2','subnet-0e07763243186cac5','subnet-02c5fea7482f804fb','subnet-0291499fd1718ee01','subnet-01c4667ad446c8337','subnet-025e9043c44114baa']";;
+        1) subnets="['subnet-0170ee5ccdd7e7823','subnet-0d50cac95bebb5a6e','subnet-0094864467fc2e737','subnet-0daa3919d85296eb6','subnet-0ab1e11d3ed63cc97','subnet-07681ad7ce2b6c281']";;
+        2) subnets="['subnet-00de9462cf29cd3d3','subnet-06595d2851257b4df','subnet-04bbfdd9ca1b67e74','subnet-096992ef7d807f6b4','subnet-0b3d7ba41fc6278b2','subnet-0b99293450e2edb13']";;
+        3) subnets="['subnet-047f6294332aa3c1c','subnet-0c3bce80bbc2c8f1c','subnet-038c38c7d96364d7f','subnet-027a025e9d9db95ce','subnet-04d9008469025b101','subnet-02f75024b00b20a75']";;
+        *) echo >&2 "invalid subnets index"; exit 1;;
```

Comment:
this should expand to pick up #6833 and #6845, but I can do that in follow-up work if you don't want to pick that up here.  You may also want to pull in `has_variant`, e.g. from [here][1].

[1]: https://github.com/openshift/release/commit/c6c2efb3fc52907970a6e5c1cc46577aaaa55d42#diff-e9ff14653190666556f490397d19c553R396

Reply:
Good as follow-ups when this is in prod.

Reply:
I'm updating my local branch after every rebase, the final version of the PR will have those.